### ISSUE TITLE
Regression: typing.Self diagnostics

### ISSUE THREAD

**Issue by jeffective** [astral-sh/ty#1579]
### Summary

This was working in 0.0.1-alpha.25 but regressed in 0.0.1-alpha.26

```python
from typing import Self

class Encoding:
    APPLICATION_CBOR: Self

def use_encoding(encoding: Encoding):
    pass

use_encoding(Encoding.APPLICATION_CBOR)
```

```
Argument to function `use_encoding` is incorrect: Expected `Encoding`, found `typing.Self` (invalid-argument-type) [Ln 9, Col 14]
```

Playground link: https://play.ty.dev/0e180658-87a1-4b6b-83b8-ab6a34dfd8b8

### Version

0.0.1-alpha.26

**Comment by carljm** [comment]
Thanks for the minimized report!

It looks to me like this fails in the same way ty alpha.25, so I suspect that some other improvement in alpha.26 made the type visible to ty in your full codebase context?

But the root issue is #1124
